common/src/primitives/compact.rs

```diff
@@ -51,15 +52,15 @@ impl TryFrom<Compact> for Uint256 {
 }
 
 // https://github.com/bitcoin/bitcoin/blob/7fcf53f7b4524572d1d0c9a5fdc388e87eb02416/src/arith_uint256.cpp#L223
-impl From<Uint256> for Compact {
-    fn from(value: Uint256) -> Self {
-        let mut size = (value.bits() + 7) / 8;
+impl From<U256> for Compact {
+    fn from(value: U256) -> Self {
+        let mut size = (value.bits_vartime() + 7) / 8;
 
         let mut compact = if size <= 3 {
-            value.low_u64() << (8 * (3 - size))
+            value.to_words()[0] << (8 * (3 - size)) // FIXME: a problem for 32bit OS???
```

Comment:
At this point I'm getting more and more convinced that this is becoming a mess very quickly. Perhaps we should just establish our own `U256` type as a wrapper for `crypto_bigint::U256`, where inside it, we limit all these operations to what's necessary and to what works everywhere, so these mistakes can never be made.